concepts/result/about.md

```diff
@@ -0,0 +1,113 @@
+# About
+
+The `Result` type is the solution in the Elm language for error reporting and an enabler for error recovery. It is somewhat similar to the `Maybe` type, but where Maybe can be returned by functions that may fail, it doesn't tell you anything about why it failed. Result includes this context about why something failed.
+
+## Definition and type variable
+
+The `Result` type is defined as follows:
+
+```elm
+type Result error value
+  = Ok value
+  | Err error
+```
+
+This is known as a "custom type" definition in Elm terminology.
+Custom types are introduced in details in another concept of this track but we won't need it to understand specifically `Result`.
+The `error` and `value` in `OK value` and `Error error` represent a type variable, meaning it can be any type, such as `Int`, `Bool` or `String`.
+As such, a `Result String Int` is a variable that is expected to contain an `Int`, and if there is an error this will be represented as a `String`.
+
+```elm
+ageFromString : String -> Result String Int
+ageFromString input =
+  case String.toInt input of
+    Nothing ->
+      Err "That is not a number!"
+
+    Just age ->
+      if age < 0 then
+        Err "Please try again after you are born."
+
+      else if age > 135 then
+        Err "Are you some kind of turtle?"
+
+      else
+        Ok age
+```
+
+The vertical bar `|` in the type definition of `Result` means "OR".
+It indicates that it can either be `Ok` something OR be `Error` something.
+
+## Reading the content of a Result value
+
+Reading the content of a `Result` value is done via "pattern matching".
+Pattern matching is also introduced in more details in another concept, so we just focus on how pattern matching a `Result` works.
+
+```elm
+formatAge : Int -> String
+formatAge age =
+    "You are " ++ toString(age)
+
+howOldAmI : String -> String
+howOldAmI age =
+    case ageFromString age of
+        Err message -> message
+        Ok age -> formatAge age
+
+howOldAmI "Bananas"
+    --> "That is not a number!"
+
+howOldAmI "32"
+    --> "You are 32"
```

Comment:
I think the examples already look different from function definitions (you couldn't have "32" in a function definition). This is also the way that all the other example code is written, so I think we should leave it as it is.

Reply:
It tripped me up a bit, because in Haskell it would be a valid way to start a definition. Also in other languages like Elixir, so it might be confusing for other people than me.
Then maybe adding a type signature?
```elm
bananasAge : String
bananasAge = howOldAmI "Bananas"
```
If not, I might do a PR to consistently change all the example code :)